membership-common/src/main/scala/com/gu/memsub/subsv2/reads/SubJsonReads.scala

```diff
@@ -72,31 +88,23 @@ object SubJsonReads {
           (json \ "lastChangeType").validateOpt[String] |@|
           (json \ "subscriptionProductFeatures").validate[List[RestFeature]] |@|
           dates.map(_.map(_._3).sorted.reduceOption(_ orElse _).flatten) |@|
-          (json \ "ratePlanCharges").validate[NonEmptyList[ZuoraCharge]](nelReads(niceListReads(zuoraSubscriptionRatePlanChargeReads))) |@|
+          (json \ "ratePlanCharges").validate[NonEmptyList[RatePlanCharge]](nelReads(niceListReads(ratePlanChargeReads))) |@|
           dates.flatMap(_.map(_._1).sorted.headOption.fold[JsResult[LocalDate]](JsError("Missing start"))(JsSuccess(_))) |@|
           dates.flatMap(_.map(_._2).sorted.headOption.fold[JsResult[LocalDate]](JsError("Missing end"))(JsSuccess(_)))
-      )(SubscriptionZuoraPlan).withTrace("low-level-plan")
-    }
-  }
-
-  val subZuoraPlanListReads: Reads[List[SubscriptionZuoraPlan]] = new Reads[List[SubscriptionZuoraPlan]] {
-    override def reads(json: JsValue): JsResult[List[SubscriptionZuoraPlan]] = {
-      (json \ "ratePlans").validate[List[SubscriptionZuoraPlan]](niceListReads(commonZuoraPlanReads))
+      )(RatePlan).withTrace("low-level-plan")
     }
   }
 
-  val multiSubJsonReads: Reads[List[JsValue]] = new Reads[List[JsValue]] {
-    override def reads(json: JsValue): JsResult[List[JsValue]] = json \ "subscriptions" match {
-      case JsDefined(JsArray(subs)) => JsSuccess(subs.toList)
-      case _ => JsError("Found no subs")
-    }
+  val multiSubJsonReads: Reads[List[Subscription]] = new Reads[List[Subscription]] {
+    override def reads(json: JsValue): JsResult[List[Subscription]] =
+      (json \ "subscriptions").validate(Reads.traversableReads[List, Subscription](implicitly, subscriptionReads))
   }
 
   private val lenientDateTimeReader: Reads[DateTime] =
     JodaReads.DefaultJodaDateTimeReads orElse Reads.IsoDateReads.map(new DateTime(_))
 
-  val subscriptionReads: Reads[NonEmptyList[RatePlan] => Subscription] = new Reads[NonEmptyList[RatePlan] => Subscription] {
-    override def reads(json: JsValue): JsResult[NonEmptyList[RatePlan] => Subscription] = {
+  val subscriptionReads: Reads[Subscription] = new Reads[Subscription] {
+    override def reads(json: JsValue): JsResult[Subscription] = {
```

Comment:
this PR meant we managed to get rid of the whole "Reads of a function type" that enabled us to do two layers of deserialisation on the plans.